src/ui/public/modals/confirm_modal.html

```diff
@@ -1,32 +1,10 @@
-<div class="kuiModal" style="width: 450px" data-test-subj="confirmModal">
-  <div class="kuiModalHeader" ng-if="title">
-    <div class="kuiModalHeader__title" data-test-subj="confirmModalTitleText">
-      {{title}}
-    </div>
-  </div>
-  <div class="kuiModalBody">
-    <div
-      class="kuiModalBodyText"
-      data-test-subj="confirmModalBodyText"
-    >
-      {{message}}
-    </div>
-  </div>
-
-  <div class="kuiModalFooter">
-    <button
-      class="kuiButton kuiButton--hollow"
-      data-test-subj="confirmModalCancelButton"
-      ng-click="onCancel()"
-    >
-      {{cancelButtonText}}
-    </button>
-    <button
-      class="kuiButton kuiButton--primary"
-      data-test-subj="confirmModalConfirmButton"
-      ng-click="onConfirm()"
-    >
-      {{confirmButtonText}}
-    </button>
-  </div>
-</div>
+<confirm-modal
```

Comment:
yes, the angular directive is globally registered in the
body of `react_component.js`:

```javascript
app.directive('confirmModal', function (reactDirective) {
  return reactDirective(KuiConfirmModal);
});
```

Since javascript modules are singletons, it will only be executed once, no
matter how often that file is imported. If a directive of the same name is
registered again in some other place, the order of execution of the two
statements matters. The last directive definition wins.

All directives are global in an angular 1 app (as are services, filters,
etc...). I don't think there's a way around that. That's one of the things
angular 2 does better (and react, of course). The import and the usage will
always be in separate files, because you cannot import anything in an angular
template.